src/test/integration/statistic/test_update.py

```diff
@@ -185,3 +189,36 @@ def test_fetch_mitigations(self):
 
     def test_known_vulnerabilities_works(self):
         self.assertEqual(self.updater.get_known_vulnerabilities_stats(), {'known_vulnerabilities': []})
+
+
+class TestStatisticWithDb(TestStatisticBase):
+    def setUp(self):
+        super().setUp()
+        self.db_backend_interface = BackEndDbInterface(config=self.config)
+
+    def tearDown(self):
+        self.db_backend_interface.client.drop_database(self.config.get('data_storage', 'main_database'))
+        self.db_backend_interface.shutdown()
+        super().tearDown()
+
+    def test_get_executable_stats(self):
+        for i, file_str in enumerate([
+            'ELF 64-bit LSB executable, x86-64, dynamically linked, for GNU/Linux 2.6.32, not stripped',
+            'ELF 32-bit MSB executable, MIPS, MIPS32 rel2 version 1 (SYSV), statically linked, not stripped',
+            'ELF 64-bit LSB executable, x86-64, (SYSV), corrupted section header size',
+            'ELF 64-bit LSB executable, aarch64, dynamically linked, stripped',
+            'ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, stripped'
+        ]):
+            fo = create_test_file_object()
+            fo.processed_analysis['file_type'] = {'full': file_str}
+            fo.uid = str(i)
+            self.db_backend_interface.add_file_object(fo)
+
+        stats = self.updater.get_executable_stats().get('executable_stats')
+        expected = [
+            ('big endian', 1, 0.25), ('little endian', 3, 0.75), ('stripped', 1, 0.25), ('not stripped', 2, 0.5),
+            ('32-bit', 1, 0.25), ('64-bit', 3, 0.75), ('dynamically linked', 2, 0.5), ('statically linked', 1, 0.25),
+            ('missing section info', 1, 0.25)
+        ]
+        for expected_item, stat_item in zip(expected, stats):
```

Comment:
This construction, I don't like. No for real: Can you not do sth. like 
```python
for label, count, percentage in expected:
    _, db_count, db_percentage, __ = stats[label]
    assert count == db_count
    assert percentage == db_percentage
```
I get that that's less compact compared to your solution, but it might be more readable.